pypeit/scripts/view_fits.py

```diff
@@ -55,44 +57,57 @@ def main(args):
-            msgs.error('You cannot specify --proc and --det mosaic, since --mosaic can only '
-                       'display the raw image mosaic')
+#        if args.proc and args.det == 'mosaic':
+#            msgs.error('You cannot specify --proc and --det mosaic, since --mosaic can only '
+#                       'display the raw image mosaic')
         if args.exten is not None and args.det == 'mosaic':
             msgs.error('You cannot specify --exten and --det mosaic, since --mosaic displays '
                        'multiple extensions by definition')
 
-
         if args.exten is not None:
             hdu = io.fits_open(args.file)
             img = hdu[args.exten].data
+            hdu.close()
         else:
             spectrograph = util.load_spectrograph(args.spectrograph)
             bad_read_message = 'Unable to construct image due to a read or image processing ' \
                                'error.  Use case interpreted from command-line inputs requires ' \
                                'a raw image, not an output image product from pypeit.  To show ' \
                                'a pypeit output image, specify the extension using --exten.  ' \
                                'Use --list to show the extension names.'
+            if 'mosaic' in args.det:
+                mosaic = True
+                _det = spectrograph.default_mosaic 
+                if _det is None:
+                    msgs.error(f'{args.spectrograph} does not have a known mosaic')
+            else:
+                try:
+                    _det = tuple(int(d) for d in args.det)
+                except:
+                    msgs.error(f'Could not convert detector input to integer.')
+                mosaic = len(_det) > 1
+                if not mosaic:
+                    _det = _det[0]
+
             if args.proc:
                 # Use the biasframe processing parameters because processing
                 # these frames is independent of any other frames (ie., does not
                 # perform bias subtraction or flat-fielding)
                 par = spectrograph.default_pypeit_par()['calibrations']['biasframe']
                 try:
-                    img = buildimage.buildimage_fromlist(spectrograph, int(args.det), par,
-                                                        [args.file]).image
+                    img = buildimage.buildimage_fromlist(spectrograph, _det, par,
+                                                         [args.file], mosaic=mosaic).image
                 except Exception as e:
                     msgs.error(bad_read_message 
                                + f'  Original exception -- {type(e).__name__}: {str(e)}')
             else:
-                det = None if args.det == 'mosaic' else int(args.det)
                 try:
-                    img = spectrograph.get_rawimage(args.file, det)[1]
+                    img = spectrograph.get_rawimage(args.file, _det)[1]
```

Comment:
It seems like these try: excepts: should be replaced by actual sensible failure messages in the relevant reading routines, but that is probably beyond the scope of this PR.